alacritty/src/display/mod.rs

```diff
@@ -618,10 +632,36 @@ impl Display {
 
         // Resize terminal.
         terminal.resize(self.size_info);
+    }
+
+    /// Update the state of the actual renderer.
+    ///
+    /// NOTE: the update to the renderer is split from the display update on purpose, since
+    /// on some platforms, like Wayland, resize and other OpenGL operations must be performed
+    /// right before rendering, otherwise they could lock the back buffer resulting in
+    /// rendering with the buffer of old size.
```

Comment:
Love this comment btw, I wasn't completely following the original issue and this makes it very clear.